Sources/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -66,11 +66,23 @@ class PurchasesOrchestrator {
     private let beginRefundRequestHelper: BeginRefundRequestHelper
     private let lock = NSRecursiveLock()
 
```

Comment:
I think it's a great workaround, but another possible solution could be to create an object that encapsulates all this logic, something like `StoreKit2Listeners` without `@available`, and it's only initialised if StoreKit2 is available

Reply:
`StoreKit2TransactionListener` is already a wrapper, so that feels like creating a wrapper for a wrapper :P